Update README.md

Treevuhh opened the pull request:
added further details per @DecentralizedDan request. Terminology and structure will need changing as needed.

Treevuhh committed:
Update README.md

added further details. Terminology and structure will need changing as needed.

DecentralizedDan commented:
@Treevuhh This looks good! I do have a couple notes:
* The first version of this diagram is factually wrong. I said there was a denylist and there is not.
  * The language `Is URL allow/denylisted **on the server**` is unhelpful bc *everything* is on a server (vs eg a browser)
  * I suggest `Is URL allowlisted by guild owner?` for accuracy
  * The arrow going down the right side from that block must be deleted, bc again, no denylist (yet!)
    * We must delete `URL on denylist`, too
  * ...and you must delete the box on the lower left side, `URL on allowlist` bc it's redundant
    * Arrow will go straight to `Bouncer leaves message untouched`

Treevuhh committed:
In response to Dan's comment

let me know if this fixes everything so far. I am still learning about how to make it more visually easy to digest but this will have to do for now.

DecentralizedDan commented:
LGTM!
